books/RayTracingInOneWeekend.html

```diff
@@ -2779,6 +2782,239 @@
 
 
 
+Creating a Scene
+====================================================================================================
+Sometimes you just have to make a scene. At this point, our `main.cc` has a standalone function to
+recursively send rays throughout the scene, and our `main()` function has the world we're rendering,
+a camera to describe the way we set up rays, plus an inline block of code to generate rays for the
+image. Things have gotten complex enough to warrant a refactoring.
+
+
+The scene Class
+----------------
+<div class='together'>
+Let's gather together all of the components of a scene into a single class that manages everything.
+Naturally, we'll name this the `scene` class.
+
+    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ C++
```

Comment:
split up the inline source of the scene class.

Something like
1. Scene class and variables
2. ray_color
3. render

While everything in this class is collated from elsewhere, if you have a really long wall of inline source, that encourages the reader to just copy/paste the code over. Which will reduce the educational value.

Have writable chunks

Reply:
Cutting and pasting is exactly the point of this step. All of the code already exists in `main.cc`, and the work routines of the `scene` class are moved from elsewhere. The point of the listing is to show where these old chunks land in the new class.